src/optimizer/filter_combiner.cpp

```diff
@@ -595,40 +596,110 @@ TableFilterSet FilterCombiner::GenerateTableScanFilters(const vector<idx_t> &col
 
 			//! Check if values are consecutive, if yes transform them to >= <= (only for integers)
 			// e.g. if we have x IN (1, 2, 3, 4, 5) we transform this into x >= 1 AND x <= 5
-			if (!type.IsIntegral()) {
-				continue;
-			}
+			bool can_simplify_in_to_range = true;
+			if (type.IsIntegral()) {
+				for (idx_t i = 1; i < func.children.size(); i++) {
+					auto &const_value_expr = func.children[i]->Cast<BoundConstantExpression>();
+					D_ASSERT(!const_value_expr.value.IsNull());
+					in_values.push_back(const_value_expr.value.GetValue<hugeint_t>());
+				}
 
-			for (idx_t i = 1; i < func.children.size(); i++) {
-				auto &const_value_expr = func.children[i]->Cast<BoundConstantExpression>();
-				D_ASSERT(!const_value_expr.value.IsNull());
-				in_values.push_back(const_value_expr.value.GetValue<hugeint_t>());
-			}
-			if (in_values.empty()) {
-				continue;
-			}
+				if (in_values.empty()) {
+					continue;
+				}
 
-			sort(in_values.begin(), in_values.end());
+				sort(in_values.begin(), in_values.end());
 
-			bool can_simplify_in_clause = true;
-			for (idx_t in_val_idx = 1; in_val_idx < in_values.size(); in_val_idx++) {
-				if (in_values[in_val_idx] - in_values[in_val_idx - 1] > 1) {
-					can_simplify_in_clause = false;
-					break;
+				for (idx_t in_val_idx = 1; in_val_idx < in_values.size(); in_val_idx++) {
+					if (in_values[in_val_idx] - in_values[in_val_idx - 1] > 1) {
+						can_simplify_in_to_range = false;
+						break;
+					}
 				}
 			}
-			if (!can_simplify_in_clause) {
-				continue;
+			// if we are not consecutive or we are VARCHAR, then we can still push an OR zone_map filter
+			if (!can_simplify_in_to_range && type.IsIntegral()) {
+				auto or_filter = make_uniq<ConjunctionOrFilter>();
+				for (idx_t in_val_idx = 1; in_val_idx < func.children.size(); in_val_idx++) {
+					D_ASSERT(func.children[in_val_idx]->type == ExpressionType::VALUE_CONSTANT);
+					auto &const_val = func.children[in_val_idx]->Cast<BoundConstantExpression>();
+					auto zone_map_filter = make_uniq<ZoneMapFilter>();
+					zone_map_filter->child_filter =
+					    make_uniq<ConstantFilter>(ExpressionType::COMPARE_EQUAL, const_val.value);
+					or_filter->child_filters.push_back(std::move(zone_map_filter));
+				}
+				table_filters.PushFilter(column_index, std::move(or_filter));
+				break;
```

Comment:
Do we need to `break` out of the loop here? If so, can you add a comment why?

Reply:
Ah, we don't. Good catch